### PR TITLE
Closes #202 | Implement dataloader for WIT

### LINKED ISSUE #202: Create dataset loader for WIT
Dataloader name: `wit/wit.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?wit

|     Dataset| wit |
|-------------|---|
| Description | Wikipedia-based Image Text (WIT) Dataset is a large multimodal multilingual dataset. WIT is composed of a curated set of 37.6 million entity rich image-text examples with 11.5 million unique images across 108 Wikipedia languages. 9 languages are spoken in Southeast Asia region |
| Subsets | - |
| Languages | ceb, fil, ind, jav, zlm, mya, tha, vie, war |
| Tasks | Image-to-Text Generation, Text-to-Image Generation |
| License | Creative Commons Attribution Share Alike 3.0 (cc-by-sa-3.0) |
| Homepage | https://github.com/google-research-datasets/wit/blob/main/DATA.md |
| HF URL | https://huggingface.co/datasets/google/wit |
| Paper URL | https://dl.acm.org/doi/abs/10.1145/3404835.3463257 |

### PR DESCRIPTION
**Pull request by raileymontalan**
Closes [#202 ](https://github.com/SEACrowd/seacrowd-datahub/issues/202).

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/wit/wit.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/wit/wit.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/wit/wit.py
Comment (ljvmiranda921) [inline seacrowd/sea_datasets/wit/wit.py:94]: Let's remove this if we're not using it

```suggestion
```

**Review by ljvmiranda921 — COMMENTED** [review]
Just missing an `__init__.py` file and some nits
But other than that, the dataloader works properly :) Thank you!

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @raileymontalan, I read [Section 3.2 of the dataset's paper](https://arxiv.org/pdf/2103.01913.pdf). This explains why there are multiple texts related to the image.

For the `texts` in the `seacrowd_imtext` schemas, could we please use the value of `caption_reference_description`? If it's `None`, then we use the value of `caption_attribution_description`. If it's also `None`, then we use the value of `caption_alt_text_description`.

In addition, could you also please note down which description is used for the data instance as a value of the `context` in the `seacrowd_imtext`?

**Comment by ljvmiranda921** [comment]
@holylovenia I think that logic works well and would love to see this change 👍 . Also stressing that the `context` is kinda important here because they relate to the informativeness of the image caption/description.

**Review by holylovenia — CHANGES_REQUESTED** [review]
Hi @raileymontalan, could you please remove all changes aside from those applied to `seacrowd/sea_datasets/wit/*`?

**Comment by raileymontalan** [comment]
Unnecessary commits have been removed. @holylovenia

**Review by holylovenia — APPROVED** [review]
Thanks for the changes, @raileymontalan! The dataloader works well on my side. One last request:

Could you please append this to `_DESCRIPTION`?

"Since the dataset contains multiple references, following Section 3.2 of the dataset's paper, the `seacrowd_imtext` subsets specify which reference is used for each data instance's `texts` via `context` in `metadata`."

Then, @ljvmiranda921 can merge this dataloader after he gives his approval.

**Comment by raileymontalan** [comment]
> Thanks for the changes, @raileymontalan! The dataloader works well on my side. One last request:
> 
> Could you please append this to `_DESCRIPTION`?
> 
> "Since the dataset contains multiple references, following Section 3.2 of the dataset's paper, the `seacrowd_imtext` subsets specify which reference is used for each data instance's `texts` via `context` in `metadata`."
> 
> Then, @ljvmiranda921 can merge this dataloader after he gives his approval.

Hi @holylovenia, added to the description. Thanks!

**Review by holylovenia — APPROVED** [review]
Thank you so much, @raileymontalan! 🙏 Merging now.

**Pull request merged**
